python/tank/context.py

```diff
@@ -737,10 +737,133 @@ def from_entity(tk, entity_type, entity_id):
         
         context.update(entity_context)
 
+        if entity_type == "Project":
```

Comment:
Did this change cause any tests to fail? If it didn't, then that's probably an indication of a missing test that should be added.